Makefile

```diff
@@ -36,14 +36,14 @@
 #
 # List of .NET assemblies that we can guarantee exist
 # OpenRA.Game.dll is a harmless false positive that we can ignore
-WHITELISTED_OPENRA_ASSEMBLIES = OpenRA.Game.exe OpenRA.Utility.exe OpenRA.Platforms.Default.dll OpenRA.Mods.Common.dll OpenRA.Mods.Cnc.dll OpenRA.Mods.D2k.dll OpenRA.Game.dll
+WHITELISTED_OPENRA_ASSEMBLIES = Game.exe Utility.exe OpenRA.Platforms.Default.dll OpenRA.Mods.Common.dll OpenRA.Mods.Cnc.dll OpenRA.Mods.D2k.dll OpenRA.Game.dll
 
 # These are explicitly shipped alongside our core files by the packaging script
 WHITELISTED_THIRDPARTY_ASSEMBLIES = ICSharpCode.SharpZipLib.dll FuzzyLogicLibrary.dll Eluant.dll BeaconLib.dll Open.Nat.dll SDL2-CS.dll OpenAL-CS.Core.dll DiscordRPC.dll Newtonsoft.Json.dll
 
 # These are shipped in our custom minimal mono runtime and also available in the full system-installed .NET/mono stack
 # This list *must* be kept in sync with the files packaged by the AppImageSupport and OpenRALauncherOSX repositories
-WHITELISTED_CORE_ASSEMBLIES = mscorlib.dll System.dll System.Configuration.dll System.Core.dll System.Numerics.dll System.Security.dll System.Xml.dll Mono.Security.dll netstandard.dll
+WHITELISTED_CORE_ASSEMBLIES = mscorlib.dll System.dll System.Configuration.dll System.Core.dll System.Numerics.dll System.Security.dll System.Xml.dll Mono.Security.dll netstandard.dll Microsoft.Win32.Registry.dll System.Security.AccessControl.dll System.Security.Principal.Windows.dll System.Security.AccessControl.dll System.Xml.Linq.dll System.Runtime.Serialization.dll System.Security.Principal.Windows.dll
```

Comment:
This doesn't appear to be enough to cover netcore:
```
paul@Ragnarok:OpenRA+pr/17989$ make check

Compiling in debug mode with dotnet...
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  OpenRA.Game -> /Users/paul/src/OpenRA/OpenRA.Game.dll
  OpenRA.Launcher -> /Users/paul/src/OpenRA/OpenRA.dll
  Mono: 
  OpenRA.Platforms.Default -> /Users/paul/src/OpenRA/OpenRA.Platforms.Default.dll
  OpenRA.Server -> /Users/paul/src/OpenRA/Server.dll
  OpenRA.Utility -> /Users/paul/src/OpenRA/Utility.dll
  OpenRA.Mods.Common -> /Users/paul/src/OpenRA/mods/common/OpenRA.Mods.Common.dll
  OpenRA.Mods.D2k -> /Users/paul/src/OpenRA/mods/d2k/OpenRA.Mods.D2k.dll
  OpenRA.Mods.Cnc -> /Users/paul/src/OpenRA/mods/common/OpenRA.Mods.Cnc.dll
  OpenRA.Test -> /Users/paul/src/OpenRA/test/OpenRA.Test.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:31.69

Checking runtime assemblies...
error: The following assemblies are referenced but not whitelisted:
   System.Private.CoreLib.dll
   Utility.dll
   System.Runtime.dll
   System.Collections.dll
   System.Runtime.Extensions.dll
   System.IO.FileSystem.dll
   System.Linq.dll
   System.Console.dll
   System.Threading.dll
   System.Text.RegularExpressions.dll
   System.Diagnostics.Process.dll
   System.ComponentModel.Primitives.dll
   System.Text.Encoding.Extensions.dll
   Microsoft.Win32.Primitives.dll
   System.Memory.dll
   System.Runtime.InteropServices.dll
   System.Collections.Concurrent.dll
   System.ComponentModel.TypeConverter.dll
   System.Runtime.Numerics.dll
   System.Buffers.dll
   System.Threading.Thread.dll
   System.Security.Cryptography.Algorithms.dll
   System.Security.Cryptography.Primitives.dll
   System.Threading.Tasks.dll
   System.Net.Primitives.dll
   System.Linq.Expressions.dll
   System.Runtime.Serialization.Formatters.dll
   System.Diagnostics.TraceSource.dll
   System.ObjectModel.dll
   System.Security.Claims.dll
   System.Security.Principal.dll
   System.Collections.NonGeneric.dll
```

I think it probably makes sense to define different lists for mono vs core, and pick the appropriate one at check-time.